File: apps/ios/nearby/Services/Zk/ZkEphemeral.swift
import Foundation
import LeanSui

/// Data model representing an active ephemeral cryptographic session for zkLogin.
///
/// `ZkEphemeral` holds the secure hardware-backed signing account and metadata
/// required to construct and verify the zero-knowledge login signature.
struct ZkEphemeral {
  /// The local hardware-bound ephemeral signing account (held in the Secure Enclave).
  let account: Account

  /// The maximum blockchain epoch for which this session key remains valid.
  let maxEpoch: UInt64

  /// Cryptographic salt/randomness generated for nonce blinding.
  let randomness: [UInt8]

  /// The base64URL-encoded zkLogin nonce bound to this session's parameters.
  let nonce: String
}

File: apps/android/app/src/main/java/com/variance/nearby/services/zk/ZkEphemeral.kt
package com.variance.nearby.services.zk

import com.variance.nearby.leansui.Account

/**
 * Data model representing an active ephemeral cryptographic session for zkLogin.
 *
 * `ZkEphemeral` holds the secure hardware-backed signing account and metadata
 * required to construct and verify the zero-knowledge login signature.
 *
 * @param account The local hardware-bound ephemeral signing account (held in the StrongBox HSM).
 * @param maxEpoch The maximum blockchain epoch for which this session key remains valid.
 * @param randomness Cryptographic salt/randomness generated for nonce blinding.
 * @param nonce The base64URL-encoded zkLogin nonce bound to this session's parameters.
 */
class ZkEphemeral(
    val account: Account,
    val maxEpoch: Long,
    val randomness: ByteArray,
    val nonce: String,
)
